tensorflow_addons/metrics/r_square_test.py

```diff
@@ -81,6 +81,20 @@ def test_r2_random_score(self):
         # Check results
         self.check_results(r2_obj, 0.7376327)
 
+    def test_r2_random_score_with_sample_weight(self):
+        actuals = tf.constant([35, 8000, 9, 11.5], dtype=tf.float32)
+        preds = tf.constant([2, 50, 500, 3.12], dtype=tf.float32)
+        sample_weight = tf.constant([2, 5, 1, 8], dtype=tf.float32)
+        actuals = tf.cast(actuals, dtype=tf.float32)
+        preds = tf.cast(preds, dtype=tf.float32)
+        sample_weight = tf.cast(sample_weight, dtype=tf.float32)
+        # Initialize
+        r2_obj = self.initialize_vars()
+        # Update
+        self.update_obj_states(r2_obj, actuals, preds, sample_weight=sample_weight)
+        # Check results
+        self.check_results(r2_obj, -0.4431257)
```

Comment:
That's a pretty nice move to align ourselves on scikit-learn on this. What would be nice it to run this test multiple times with random values and run the scikit-learn function to compare with your implementation. In this way we make sure we align with scikit-learn on all types of inputs and that it will stay that way in the future.

To be able to use scikit-learn in the tests, just add scikit-learn in this file: https://github.com/tensorflow/addons/blob/master/tools/install_deps/pytest.txt you can then re-run the tests with docker on cpu to check that everything works.

Reply:
Very much agree that these are good ideas. Will update the PR.

Reply:
This wasn't as straight forward as I thought. I noticed that this metric only agrees with sklearn for `y_true.shape == (x,)` or `y_true.shape == (x, 1)`, in other words it doesn't properly support multivariate regression. I feel like this is a separate issue, though, and should be dealt with in a separate PR. There are also some other subtle differences, like sklearn returning 0.0 instead of -inf in some cases: https://github.com/scikit-learn/scikit-learn/blob/95d4f0841/sklearn/metrics/_regression.py#L604

I thought the test case you described of running the test multiple times with random values was a good fit for using [hypothesis](https://hypothesis.readthedocs.io/en/latest/), so I played around with that. This can be replaced with another way of generating test data, though.

Reply:
Also noticing that hypothesis broke the build, so I might have to reconsider.

Reply:
> I feel like this is a separate issue, though, and should be dealt with in a separate PR. 

Sure thing. Let's keep this pull request small and focused. 

Can you try to make it work without hypothesis? I'm not against using it, but we have many changes for the test suite planned in the coming weeks, so it should be easier if there are less dependencies. I'll merge it as soon as the tests are passing when comparing to the scikit learn implementation.

Also could you open an issue concerning the discrepancy between scikit-learn and tf addons for the arrays of dimension 2? This will be easier to track the progress on this bug after this pull request is merged. @SSaishruthi can then look into it whenever he/she has the time.

Reply:
I do have some plans for how to make it work for vector labels, and intend to implement a fix. There are some design choices in that regard that I might want a discussion with you (I guess both you and @SSaishruthi). An issue might be a good place to do that. You can expect one from me soon.

Will definitely make it work without hypothesis. It is a powerful testing library, but it likely shouldn't be pulled in as an afterthought. It also means I don't have to figure out why it failed and how to fix it 😅.